### PR TITLE
DSG-1913/Upgrade-Java-version

### PR DESCRIPTION
**Pull request by afcalderonm**
## Description
Upgrade java version to 17

## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -210,10 +210,10 @@ set -- \
...
 
 # Stop when "xargs" is not available.
-if ! command -v xargs >/dev/null 2>&1
-then
-    die "xargs is not available"
-fi
+##if ! command -v xargs >/dev/null 2>&1
+##then
+##    die "xargs is not available"
+##fi
```
Comment (asultan001) [inline gradlew:216]: Why is this commented out? should this be removed all together?

File: service/src/main/resources/application.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@ spring:
       write-dates-as-timestamps: false
     deserialization:
       adjust_dates_to_context_time-zone: false
+  config:
+    import: sm://
```
Comment (asultan001) [inline service/src/main/resources/application.yml:24]: why is this needed?

File: gradlew (hunk 1 above)
Comment (afcalderonm) [reply gradlew:216]: yeah it needs to be removed because openjdk 17 does not have xargs by default the cloud worker always fails by that validation

File: service/src/main/java/io/nuvalence/workmanager/service/utils/RequestContextTimestamp.java [hunk 3]
```diff
@@ -14,8 +14,8 @@
 public class RequestContextTimestamp {
     private final OffsetDateTime currentTimestamp;
 
-    public RequestContextTimestamp(final Clock requestScopeClock) {
-        this.currentTimestamp = OffsetDateTime.now(requestScopeClock);
+    public RequestContextTimestamp() {
+        this.currentTimestamp = OffsetDateTime.now(Clock.systemUTC());
```
[the remark this answers is not in the document]
Comment (afcalderonm) [reply service/src/main/java/io/nuvalence/workmanager/service/utils/RequestContextTimestamp.java:18]: the way the clock was implemented before work for java 11, it was creating a separated Bean base on the java.time.Clock class but in java 17 there was a change that does not allow to access some private, final or non-visible classes, to it was failing to access to some classes of the Clock library and it was failing to create the Bean, i made and  analysis and found out that that bean was just returning the Clock.systemUTC(). so i remove that class and just set the Clock.systemUTC() directly, i talk with Dan who was the person to do that implementation and he says it is OK

File: gradlew [hunk 4]
```diff
@@ -209,12 +209,6 @@ set -- \
         org.gradle.wrapper.GradleWrapperMain \
         "$@"
 
-# Stop when "xargs" is not available.
-if ! command -v xargs >/dev/null 2>&1
-then
-    die "xargs is not available"
-fi
-
```
Comment (asultan001) [inline gradlew:217]: why was this removed?

Comment (afcalderonm) [reply gradlew:217]: these are  the same lines that was commented in the previous conversation

Comment (asultan001) [reply gradlew:217]: If it is not available in openjdk17 is there an alternative? the next block seems to indicate that we are uxing xargs to parse quoted args and does a bunch of work.

Comment (afcalderonm) [reply gradlew:217]: yeah there is an alternative when we can install from other source before calling the gradlew, but we are not really using any quoted args that needs xargs and it is also removed in the other services, so i dont we really need to do it at least at this moment

Comment (JayStGelais) [reply gradlew:217]: I'm not fond of manually editing the gradlew shell script. This should be generated by the wrapper task and we shouldn't have to touch it.

**Review by JayStGelais — APPROVED** [review]
